lib/Doctrine/ORM/Repository/InvalidMagicMethodCall.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Repository;
+
+use Doctrine\ORM\RepositoryException;
+
+final class InvalidMagicMethodCall extends \InvalidArgumentException implements RepositoryException
```

Comment:
I made the exception final

Reply:
I removed the `Repository` bit from the name since it already is in the namespace.

Reply:
Was `InvalidArgumentException` chosen instead of `BadMethodCallException` because of BC or because magic methods are getting the method name as an argument?

Reply:
I don't know, I copy pasted a snippet from @Ocramius but `BadMethodCallException` might be better indeed: it's not about the arguments being invalid, it's about the field not existing.